rj_gameplay/rj_gameplay/tactic/line_tactic.py

```diff
@@ -1,37 +1,39 @@
-import stp
-from rj_gameplay.role import dumb_move
 from typing import List, Tuple
+
+import numpy as np
+import stp
 from rj_msgs.msg import RobotIntent
 
+from rj_gameplay.role import dumb_move
+
 
 class LineTactic(stp.tactic.Tactic):
-    """Tactic for line up play that puts all six robots in a line on the left of the field."""
+    """Tactic for line up play that puts chosen robots in a line from start point to end point selected.
+    Assuming that the liners are more than 2"""
 
-    def __init__(self, world_state: stp.rc.WorldState, num_lined_up: int):
+    def __init__(
+        self,
+        world_state: stp.rc.WorldState,
+        num_liners: int,
+        start: np.ndarray,
+        end: np.ndarray,
+    ):
         super().__init__(world_state)
 
         # compute move points
-        # access the length and the width of the field (plus the border), and starts robots there
-        start = (
-            (world_state.field.width_m + world_state.field.border_m) / 2,
-            (world_state.field.length_m - world_state.field.border_m) / 2,
-        )
-        dy = 0.5
-        # TODO: make the # here a param instead of hardcoding for same reason as above TODO
-        self.move_points = [(start[0], start[1] + i * dy) for i in range(num_lined_up)]
-
-
-        # request closest robot every pt
-        for pt in self.move_points:
-            # for some reason stp.role doesn't need to be imported?
+        self.start = start
+        self.end = end
+        xpts = np.linspace(self.start[0], self.end[0], num_liners)
+        ypts = np.linspace(self.start[1], self.end[1], num_liners)
+        self.target_points = []
+
```

Comment:
Maybe have a check to make sure the robots can fit along the line, and throw an error if they can't. Like if the start and end point are too close or there are too many robots to fit between. You can use the diameter of the robots for calculating this.